samples/remote_attestation/common/crypto.cpp

```diff
@@ -310,3 +303,202 @@ bool Crypto::get_rsa_modulus_from_pem(
 exit_preinit:
     return ret;
 }
+
+bool Crypto::Encrypt_gcm(
+    const uint8_t* sym_key,
+    const uint8_t* data,
+    size_t data_size,
+    uint8_t* encrypted_data,
+    size_t* encrypted_data_size,
+    uint8_t* tag_output)
+{
+    mbedtls_gcm_context gcm_context;
+    unsigned char iv_str[12];
+
+    bool result = false;
+    int res = -1;
+    size_t iv_len, tag_len = 16;
+
+    mbedtls_gcm_init(&gcm_context);
+
+    memset(iv_str, 0x00, sizeof(iv_str));
+    memcpy(iv_str, "1234567890ab", sizeof("1234567890ab"));
```

Comment:
**Do not use a fixed IV for AES-GCM.** This is such a common security issue that it has prompted the design of nonce-reuse resistant variants of AES-GCM.

You've also provided a 12-byte string with a length of 16 for some reason?

Reply:
That is a bug in my code. 
I will be changing the fixed IV to be a random one as mentioned in my TODO in Description.  The recommended default length for a IV is 96 bytes as suggested by the GCM spec.  

Started this PR to get valuable feedback. So, thanks a lot.

Reply:
To add to Simon's point on why not to use a fixed iv:

GCM encryption essentially takes a counter, encrypts it and XORs it with the plain text. (i.e `ciphertext = plaintext XOR Encrypt(Counter)`).  Reusing the IV with the same key means that you can just XOR the ciphertexts together to get the XOR'd plaintext. This might be enough to figure out the plaintext outright.

There's also more complicated attacks that allow the attacker to gain a lot more information on the keys.

Reply:
@gupta-ak - Implemented random IV already.